accepted/0015-virtual-camera-support.md

```diff
@@ -17,26 +17,55 @@ Similar functionality:
 * [The Snap Camera](https://snapcamera.snapchat.com) is an example of an application that consumes input from a hardware webcam, processes the video stream, and outputs it in real time as a virtual device that appears in apps like Skype, Hangouts, or Zoom. It supports both Mac & PC.
 
 # Detailed design
-- TODO - I don't know anything about OBS internals and so don't know how to design this.
-- Platform specific device driver?
+
+- Platform specific plugins shall be created that will be added as part of the main obs build.
+- Each plugin will start a server that will be responsible of transmitting the frame data to a client process.
+- A virtualcam "Driver" (client process) will be developed for each platform to consume data incomming from this plugins.
+- Each virtual driver shall be part of the installer with a checkbox to install the virtualcam driver
+
+## Cross-platform Interfaces
+
+We can define domain interfaces in order to maintain uniformity in the platform-specific plugins (maybe someday they will not be platform specific).
+
+- **OBSVirtualCamServer**: Interface that defines the set of methods to be implemented by a OBSVirtualCam Server in order to send and receive frame data (and other metadata) to the client processes.
+- **OBSVirtualCamClient**: Interface that defines the set of methods to be implemented by a OBSVirtualCam Client in order to register with a server(obs-studio) and receive frame data.
+
+## Platform specific implementations
+
+Each platform specific implementation will be separated into 2 cmake projects (subprojects?).
+- **<Platform>OBSVirtualCam OBS Plugin**: Platform speficic obs plugin module that will be loaded by obs-studio in order to expose the virtual cam OBS Output and its controls. This will implement **OBSVirtualCamServer** and interfaces
+- **<Platform>OBSVirtualCam OBS Driver**: Platform specific driver code that will register the camera with the OS and will register with the **OBSVirtualCamServer** in order to receive data. If no connection to the server is available it should still register and render an static frame and keep waiting for the **OBSVirtualCamServer**. This will implement **OBSVirtualCamClient** interfaces.
+
+### Windows
+
+#### WindowsOBSVirtualCam OBS Module
+#### WindowsOBSVirtualCam Driver
+
+### Mac
+
+#### MacOBSVirtualCam OBS Module
+#### MacOBSVirtualCam Drivers
 
 ## Technical considerations
 
-- TODO
+- This RFC is explicitly for an OBS-specific implementation of output from OBS to a virtual camera.
+- A generic middleware solution to this problem is out of scope at the moment
+- A virtual camera needs to be registered with the system. Usually, this'll be done via the OBS installer/updater
+- The virtual device/output should be accessible by third party applications that support webcams/capture cards
 
 ## User UX
 
+- When installing OBS user should be asked if he would like to install the platform specific driver to enable virtual camera output on the system
```

Comment:
I don't think this is necessary. I think we should just install it.

Reply:
fixed